1.sh

```diff
@@ -1,35 +1,34 @@
 #! /bin/sh
 
-if [-d $1] && [-d $2]
+if [ -d "$1" ] && [ -d "$2" ]
 then
 	echo "Dirs exist"
-elif [! -d $2] 
+elif [ ! -d "$2" ] 
 	echo "Flash path doesn't exist"
-	mkdir $2
+	mkdir "$2"
 fi
 
-for file in $1/*
+for file in "$1"/*
```

Comment:
Notice how I didn't wrap `/*` in quotes, too? That's because bash would then not have been able to interpret the asterix (glob) for Filename Expansion. Technically, it could be written as `"$1/"*`, but I found the other approach more readable.